workflow-manager/main.go

```diff
@@ -254,7 +389,270 @@ func getReadyBatchesGS(ctx context.Context, inputBucket string) ([]string, error
+			log.Printf("ready: %s", k)
+			readyBatches = append(readyBatches, k)
+		} else {
+			log.Printf("unready: %s", k)
+		}
+	}
+	return readyBatches, nil
+}
+
+func launchAggregationJob(ctx context.Context, readyBatches []string) error {
+	if len(readyBatches) == 0 {
+		log.Printf("no batches to aggregate")
+		return nil
+	}
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return fmt.Errorf("cluster config: %w", err)
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("clientset: %w", err)
+	}
+	aggregationID := ""
+	var earliestBatch *batchPath
+	var latestBatch *batchPath
+	batchIDs := []string{}
+	batchTimes := []string{}
+
+	for _, batchName := range readyBatches {
+		batchPath, err := newBatchPath(batchName)
+		if err != nil {
+			return err
+		}
+		batchIDs = append(batchIDs, batchPath.batchID)
+		batchTimes = append(batchTimes, batchPath.batchDateString())
+
+		// All batches should have the same aggregation ID?
+		if aggregationID != "" && aggregationID != batchPath.aggregationID {
+			return fmt.Errorf("found batch with aggregation ID %s, wanted %s", batchPath.aggregationID, aggregationID)
+		}
+		aggregationID = batchPath.aggregationID
+
+		// Find earliest and latest batch in the aggregation
+		currentBatchTime, err := batchPath.batchDateTime()
+		if err != nil {
+			return err
+		}
+		if earliestBatch == nil {
+			earliestBatch = &batchPath
+		} else {
+			earliestBatchTime, err := earliestBatch.batchDateTime()
+			if err != nil {
+				return err
+			}
+			if currentBatchTime.Before(earliestBatchTime) {
+				earliestBatch = &batchPath
+			}
+		}
+		if latestBatch == nil {
+			latestBatch = &batchPath
+		} else {
+			latestBatchTime, err := latestBatch.batchDateTime()
+			if err != nil {
+				return err
+			}
+			if currentBatchTime.After(latestBatchTime) {
+				latestBatch = &batchPath
+			}
+		}
+	}
+
+	args := []string{
+		"aggregate",
+		"--aggregation-id", aggregationID,
+		"--aggregation-start", earliestBatch.batchDateString(),
+		"--aggregation-end", latestBatch.batchDateString(),
+	}
+	for _, batchID := range batchIDs {
+		args = append(args, "--batch-id")
+		args = append(args, batchID)
+	}
+	for _, batchTime := range batchTimes {
+		args = append(args, "--batch-time")
+		args = append(args, batchTime)
+	}
+
+	jobName := strings.ReplaceAll(fmt.Sprintf("a-%s", earliestBatch.batchDateString()), "/", "-")
+
+	log.Printf("starting aggregation job with args %s", args)
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      jobName,
+			Namespace: *k8sNS,
+		}, Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					ServiceAccountName: *k8sServiceAccount,
+					RestartPolicy:      "Never",
+					Containers: []corev1.Container{
+						{
+							Args:            args,
+							Name:            "facile-container",
+							Image:           "us.gcr.io/prio-bringup-290620/prio-facilitator:latest",
```

Comment:
Just noting that this is hardcoded to a different image name than the one for intake jobs. I'm fine landing as-is; we need to land a PR to parameterize these, and that doesn't need to block this change.

Reply:
Good catch. That's the image in my container registry I've been using. Yes, let's fix that when we parameterize the container image.